QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,132 @@
+This document is a declaration of software quality for the `rcutils` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+# `rcutils` Quality Declaration
+
+The package `rcutils` claims to be in the **Quality Level 1** category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 1 in REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+## Version Policy
+
+### Version Scheme
+
+`rcutils` uses `semver` according to the recommendation for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#versioning), and is at or above a stable version, i.e. `>= 1.0.0`.
+
+### API Stability Within a Released ROS Distribution
+
+`rcutils` will not break public API within a released ROS distribution, i.e. no major releases once the ROS distribution is released.
+
+### ABI Stability Within a Released ROS Distribution
+
+`rcutils` contains C code and therefore must be concerned with ABI stability, and will maintain ABI stability within a ROS distribution.
+
+### Public API Declaration
+
+All symbols in the installed headers are considered part of the public API.
+
+All installed headers are in the `include` directory of the package, headers in any other folders are not installed and considered private.
+
+Additionally, there are generated header files which are installed and therefore part of the public API.
+The source templates for these generated headers are in the `resource` folder.
+
+## Change Control Process
+
+TODO fix link
+
+`rcutils` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#change_control_process).
```

Comment:
The ROS 2 developer guide needs a common change control process section for us to link to and not repeat ourselves too much.

Reply:
Although it's a good idea to have the common rules for versioning/documentation in the developer guide, I think that the quality declaration document should be as self contained as possible and at least contain a brief summary of those guidelines. That would avoid users jumping back and forth to the developer guide when reading the packages documentation.

Reply:
Hmm, I don't completely agree, but I think we're mostly on the same page. I think the document should make sense on its own, but details definitely should be collected in one place if possible. What I don't want to have to do is that if we change our recommendation from 1-review into 2-reviews for each pull request that I have to touch every repository manually.

In the cases where the linked to guidelines leave some flexibility, then I think it's important for the declaration that is referencing it to clarify what it is choosing to do, e.g. if it chooses line versus branch coverage.

Reply:
Yes, didn't thought about the cases where the recommended guidelines are changed, seems good then to have a 'Recommended guidelines' document to reference